integration-libs/opf/base/root/services/opf-resource-loader.service.ts

```diff
@@ -0,0 +1,198 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { DOCUMENT, isPlatformServer } from '@angular/common';
+import { Inject, Injectable, PLATFORM_ID } from '@angular/core';
+import { ScriptLoader } from '@spartacus/core';
+
+import { throwError } from 'rxjs';
+import {
+  OpfDynamicScriptResource,
+  OpfDynamicScriptResourceType,
+} from '../model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class OpfResourceLoaderService extends ScriptLoader {
```

Comment:
[MEDIUM]

It does not need to extend `ScriptLoader`. Instead, it can simply inject `ScriptLoader` and use it.